src/bootman/kernel.c

```diff
@@ -569,6 +572,11 @@ bool boot_manager_install_kernel_internal(const BootManager *manager, const Kern
                 return false;
         }
 
+        /* Don't rewrite the same initrd to disk */
+        if (cbm_files_match(kernel->initrd_file, initrd_target)) {
+                return true;
+        }
+
```

Comment:
Yeah I think I agree. I also think the kernel will have `->initrd_file` and `->user_initrd_file` and we'll select the one to use based on NULLness